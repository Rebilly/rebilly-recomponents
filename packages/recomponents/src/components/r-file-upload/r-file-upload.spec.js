import {shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import Vue from 'vue';
import RFileUpload from './r-file-upload.vue';

describe('r-file-upload.vue', () => {
    it('should render Wrapper and match snapshot', () => {
        const wrapper = shallowMount(RFileUpload, {
            propsData: {
                multiple: true,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RFileUpload, {
            propsData: {
                multiple: false,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should handle input event', async () => {
        const wrapper = shallowMount(RFileUpload, {
            propsData: {
                multiple: true,
            },
        });

        wrapper.vm.input({
            target: {
                files: [],
            },
        });

        expect(wrapper.emitted('input')).toBeTruthy();
        expect(wrapper.emitted('close')).toBeTruthy();
    });

    it('should handle open and close', async () => {
        const wrapper = shallowMount(RFileUpload, {
            propsData: {
                multiple: true,
            },
        });

        wrapper.vm.openFileBrowser();
        wrapper.vm.setFocus();

        await Vue.nextTick();

        expect(wrapper.emitted('cancel')).toBeTruthy();
        expect(wrapper.emitted('close')).toBeTruthy();
    });
});
