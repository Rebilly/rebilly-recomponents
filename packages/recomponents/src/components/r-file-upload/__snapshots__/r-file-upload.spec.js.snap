// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-file-upload.vue should render Wrapper and match snapshot 1`] = `<div class="r-file-upload fs-block"><input type="file" multiple="multiple" class="r-file-upload-input"> </div>`;

exports[`r-file-upload.vue should render via SSR and match snapshot 1`] = `<div class="r-file-upload"><input type="file" class="r-file-upload-input"> </div>`;
